packages/compat/src/v1-config.ts

```diff
@@ -49,16 +49,12 @@ export class WriteV1Config extends Plugin {
 }
 
 function metaLoader(appName: string) {
+  // Added contentFor config-module support.
+  // ember-cli adds content in AMD module format and
+  // not as ES6 JS modules. Changed template to match ember-cli
   return `
-  let config, metaName;
-  try {
-    metaName = '${appName}/config/environment';
-    let rawConfig = document.querySelector('meta[name="' + metaName + '"]').getAttribute('content');
-    config = JSON.parse(unescape(rawConfig));
-  }
-  catch(err) {
-    throw new Error('Could not read config from meta tag with name "' + metaName + '".');
-  }
-  export default config;
+    define('${appName}/config/environment', [], function() {
```

Comment:
Although ember-cli produces AMD here, could we continue to produce ES6 modules? Or is there a technical limitation preventing us from doing so.

In general, the idea of AMD in embroider is only for legacy compatibility. If core features such as config/environment cannot be implemented in the modern way, maybe embroider is missing an important primitive.

Reply:
I think the technical limitation is related to what ember-cli returns as content for `config-module`. The Content is expected to return an export object as show [here](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/app-config.js) and replaces a placeholder in this [AMD module ](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/app-config.js).

Reply:
@stefanpenner as we discussed I updated the Code to use the immediate invocation within the ES6. Also, updated the comments.